openmc/mgxs/library.py

```diff
@@ -717,3 +725,307 @@ def load_from_file(filename='mgxs', directory='mgxs'):
 
         # Load and return pickled Library object
         return pickle.load(open(full_filename, 'rb'))
+
+    def write_mg_library(self, xs_type='macro', domain_names=None, xs_ids=None,
+                         filename='mg_cross_sections', directory='./',
+                         return_names=True):
+        """Creates a cross-section data library file for the Multi-Group
+        mode of OpenMC.
+
+        Parameters
+        ----------
+        xs_type: {'macro', 'micro'}
+            Provide the macro or micro cross section in units of cm^-1 or
+            barns. Defaults to 'macro'. If the Library object is not tallied by
+            nuclide this will be set to 'macro' regardless.
+        domain_names : Iterable of str
+            List of names to apply to the xsdata entries in the
+            resultant mgxs data file. Defaults to "set1", "set2", ...
+        xs_ids : str or Iterable of str
+            Cross section set identifier (i.e., "71c") for all
+            data sets (if only str) or for each individual one
+            (if iterable of str). Defaults to '1g'
```

Comment:
Ah, yes, I intended the `g` identifier to mean multi-group (or group-wise really), sort of like how the `c` in `70c` means continuous-energy or `t` means thermal scattering.  The `1` is just a number that can be anything and depends on the users naming scheme - maybe they'd want to do 70 because its based on ENDF-B/VII, but who knows).  I now see that `1g` may make some folks think something like "one group" instead of the above.  Therefore I've changed it to `1m` where the m is for `multi-group`.